hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -325,14 +326,15 @@ public boolean canDisableFsSnapshot(OMMetadataManager ommm) {
     return isSnapshotInfoTableEmpty;
   }
 
-  private CacheLoader<String, OmSnapshot> createCacheLoader() {
-    return new CacheLoader<String, OmSnapshot>() {
+  private CacheLoader<UUID, OmSnapshot> createCacheLoader() {
+    return new CacheLoader<UUID, OmSnapshot>() {
 
       @Nonnull
       @Override
-      public OmSnapshot load(@Nonnull String snapshotTableKey)
-          throws IOException {
+      public OmSnapshot load(@Nonnull UUID snapshotId) throws IOException {
         // Check if the snapshot exists
+        String snapshotTableKey = ((OmMetadataManagerImpl) ozoneManager.getMetadataManager()).getSnapshotChainManager()
```

Comment:
Should we add a null check here just as a sanity check?

Reply:
It makes sense to add a null check here. Added it.